shell/platform/embedder/embedder_asset_resolver.cc

```diff
@@ -0,0 +1,55 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/embedder/embedder_asset_resolver.h"
+#include "flutter/shell/platform/embedder/embedder_struct_macros.h"
+
+namespace flutter {
+
+class EmbedderAssetResolver final : public flutter::AssetResolver {
+ public:
+  explicit EmbedderAssetResolver(void* user_data,
+                                 FlutterAssetResolverGetAssetCallback get_asset)
+      : user_data_(user_data), get_asset_(get_asset) {}
+
+  ~EmbedderAssetResolver() override {}
+
+  // |AssetResolver|
+  bool IsValid() const override { return true; }
+
+  // |AssetResolver|
+  bool IsValidAfterAssetManagerChange() const override { return true; }
+
+  // |AssetResolver|
+  AssetResolverType GetType() const override {
+    return AssetResolverType::kEmbedderAssetResolver;
+  }
+
+  // |AssetResolver|
+  std::unique_ptr<fml::Mapping> GetAsMapping(
+      const std::string& asset_name) const override {
+    if (!get_asset_) {
```

Comment:
This isn't how the the other asset resolvers behave. If the asset cannot be found, `nullptr` must be returned which allows the lookup to move on to the next resolver. I don't think we have a mechanism built-in to the engine to short circuit this mechanism (nor should we have one).

Reply:
Gotcha, while the previous one was equivalent to returning nullptr, I changed it to `return nullptr` so it's more clear.